packages/react/select/src/Select.tsx

```diff
@@ -744,70 +639,83 @@ const SelectContentImpl = React.forwardRef<SelectContentImplElement, SelectConte
         content={content}
         viewport={viewport}
         onViewportChange={setViewport}
         itemRefCallback={itemRefCallback}
         selectedItem={selectedItem}
         onItemLeave={handleItemLeave}
         itemTextRefCallback={itemTextRefCallback}
+        focusSelectedItem={focusSelectedItem}
         selectedItemText={selectedItemText}
-        onScrollButtonChange={handleScrollButtonChange}
         isPositioned={isPositioned}
-        shouldExpandOnScrollRef={shouldExpandOnScrollRef}
         searchRef={searchRef}
       >
         <RemoveScroll as={Slot} allowPinchZoom>
-          <div
-            ref={setContentWrapper}
-            style={{
-              display: 'flex',
-              flexDirection: 'column',
-              position: 'fixed',
-              zIndex: contentZIndex,
+          <FocusScope
+            asChild
+            // we make sure we're not trapping once it's been closed
+            // (closed !== unmounted when animating out)
+            trapped={context.open}
+            onMountAutoFocus={(event) => {
+              // we prevent open autofocus because we manually focus the selected item
+              event.preventDefault();
             }}
+            onUnmountAutoFocus={composeEventHandlers(onCloseAutoFocus, (event) => {
+              context.trigger?.focus({ preventScroll: true });
+              event.preventDefault();
+            })}
           >
-            <FocusScope
+            <DismissableLayer
               asChild
-              // we make sure we're not trapping once it's been closed
-              // (closed !== unmounted when animating out)
-              trapped={context.open}
-              onMountAutoFocus={(event) => {
-                // we prevent open autofocus because we manually focus the selected item
-                event.preventDefault();
-              }}
-              onUnmountAutoFocus={composeEventHandlers(onCloseAutoFocus, (event) => {
-                context.trigger?.focus({ preventScroll: true });
-                event.preventDefault();
-              })}
+              disableOutsidePointerEvents
+              onEscapeKeyDown={onEscapeKeyDown}
+              onPointerDownOutside={onPointerDownOutside}
+              // When focus is trapped, a focusout event may still happen.
+              // We make sure we don't trigger our `onDismiss` in such case.
+              onFocusOutside={(event) => event.preventDefault()}
+              onDismiss={() => context.onOpenChange(false)}
             >
-              <DismissableLayer
+              <SelectPopper
                 role="listbox"
                 id={context.contentId}
                 data-state={context.open ? 'open' : 'closed'}
                 dir={context.dir}
                 onContextMenu={(event) => event.preventDefault()}
                 {...contentProps}
+                {...popperContentProps}
+                onPlaced={React.useCallback(() => setIsPositioned(true), [])}
```

Comment:
Can we remove `useCallback` now that its stable internally?

Reply:
Ah yep, completely forgot!